test/daml/Test/Pricing.daml

```diff
@@ -16,17 +16,26 @@ import Prelude hiding (max)
 
 data Instrument = USD | EUR | AMZN | APPL deriving (Show, Eq)
 
-call : Instrument -> Decimal -> Instrument -> Claim t Decimal Instrument
-call s k a = scale (O.observe s - O.pure k) $ one a
+data Observable = Spot_AMZN | Spot_APPL deriving (Show, Eq)
 
-margrabe s1 s2 a = scale (O.observe s1 - O.observe s2) $ one a
+spot : Instrument -> Observable
+spot AMZN = Spot_AMZN
+spot APPL = Spot_APPL
+spot other = error $ "disc: " <> show other
+
+call : Instrument -> Decimal -> Instrument -> Claim t Decimal Instrument Observable
+call s k a = scale (O.observe (spot s) - O.pure k) $ one a
```

Comment:
Here I am making explicit what I think was an assumption in the previous model: given a stock asset (e.g. AMZN), the `call` function is interested with observing its spot price (which is an observable, called Spot_AMZN)